zeeguu/api/endpoints/sessions.py

```diff
@@ -1,11 +1,29 @@
 import flask
+from datetime import datetime
 from flask import request, make_response
 from zeeguu.core.model import Session, User
 from zeeguu.api.utils.abort_handling import make_error
 
-from zeeguu.api.utils.route_wrappers import cross_domain, with_session
+from zeeguu.api.utils.route_wrappers import cross_domain, requires_session
 from . import api, db_session
 
+DAYS_BEFORE_EXPIRE = 30  # Days
+
+
+def is_session_valid(session_object):
+    print("----------- Validating the Session based on Last use!! -----------")
+    if session_object is None:
+        return False
+    is_session_too_old = (
+        datetime.now() - session_object.last_use
+    ).days > DAYS_BEFORE_EXPIRE
+    if is_session_too_old:
+        print("Session was too old! - logging out!")
+        db_session.delete(session_object)
+        db_session.commit()
```

Comment:
if the name of the method suggest a predicate, then we should not be doing user logout in this method. 

solution might be removing this method because it does too many things:
- checks for session object being None... that's an ocymoron
- checking for is_expired
- deleting session if too old

and introducing an `is_session_expired` method. the null check and the session deletion gcan be done more expilicitly elsewhere where they belong.  

do you see what I mean? if not, i can give a first stab to a refactoring in this direction!

Reply:
Could you check now - I have tried to make it a bit more explicit.

Reply:
TAK! It's merged and deployed. It reads quite nicely now :)